src/ConnectedMode.UnitTests/QualityProfiles/QualityProfileDownloaderTests.cs

```diff
@@ -40,51 +39,61 @@ namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.QualityProfiles
     public class QualityProfileDownloaderTests
     {
         [TestMethod]
-        public async Task UpdateAsync_AllSupportedLanguageAreChecked()
+        public void MefCtor_CheckIsExported()
         {
-            var boundProject = CreateBoundProject();
-
-            var sonarQubeService = new Mock<ISonarQubeService>();
-
-            // Note: calling the public constructor here to check that it uses
-            // the expected set of languages
-            var testSubject = new QualityProfileDownloader(sonarQubeService.Object,
-                Mock.Of<IBindingConfigProvider>(),
-                Mock.Of<IConfigurationPersister>(),
-                new TestLogger(logToConsole: true));
-            
-            await testSubject.UpdateAsync(boundProject, null, CancellationToken.None);
-
-            foreach (var langauge in Language.KnownLanguages)
-            {
-                CheckLanguageQpRequested(sonarQubeService, langauge);
-            }
-            sonarQubeService.Invocations.Should().HaveCount(Language.KnownLanguages.Count());
+            MefTestHelpers.CheckTypeCanBeImported<QualityProfileDownloader, IQualityProfileDownloader>(
+                MefTestHelpers.CreateExport<IBindingConfigProvider>(),
+                MefTestHelpers.CreateExport<IConfigurationPersister>(),
+                MefTestHelpers.CreateExport<IOutOfDateQualityProfileFinder>(),
+                MefTestHelpers.CreateExport<ILogger>());
         }
 
+        [TestMethod]
+        public void MefCtor_CheckIsSingleton()
+        {
+            MefTestHelpers.CheckIsSingletonMefComponent<QualityProfileDownloader>();
+        }
+        
         [TestMethod]
         public async Task UpdateAsync_MultipleQPs_ProgressEventsAreRaised()
         {
             // Arrange
             var logger = new TestLogger(logToConsole: true);
             var boundProject = CreateBoundProject();
 
-            var languagesToBind = new[] {
+            var languagesToBind = new[]
+            {
                 Language.Cpp,
                 Language.CSharp,
                 Language.Secrets,
                 Language.VBNET
             };
 
+            SetupAvailableLanguages(out var outOfDateQualityProfileFinderMock,
+                boundProject,
+                languagesToBind
+                    .Select(x => (x, CreateQualityProfile()))
```

Comment:
Multiple tests have this `.Select(x => (x, CreateQualityProfile()` snippet repeated to create the tuples.

It would simplify the test setup to have a version of `SetupAvailableLanguages` that just accepted a list of languages and created the tuples internally.